src/hypertable.h

```diff
@@ -26,9 +26,20 @@ typedef struct SubspaceStore SubspaceStore;
 typedef struct Chunk Chunk;
 typedef struct Hypercube Hypercube;
 
-#define TS_HYPERTABLE_HAS_COMPRESSION(ht)                                                          \
-	((ht)->fd.compressed_hypertable_id != INVALID_HYPERTABLE_ID)
+enum
+{
+	HypertableCompressionOff = 0,
+	HypertableInternalCompressionTable = 1,
+	HypertableCompressionEnabled = 2,
+};
+
+#define TS_HYPERTABLE_HAS_COMPRESSION(ht) ts_hypertable_has_compression(ht)
 
+#define TS_HYPERTABLE_HAS_COMPRESSION_ENABLED(ht)                                                  \
+	((ht)->fd.compression_state == HypertableCompressionEnabled)
+
+#define TS_HYPERTABLE_INTERNAL_COMPRESSION(ht)                                                     \
```

Comment:
Longer but less ambiguous:

```suggestion
#define TS_HYPERTABLE_IS_INTERNAL_COMPRESSION_TABLE(ht)                                                     \
```

Reply:
fixed.